### PR TITLE
Paginate contracts

### PR DESCRIPTION
**Pull request by Norbo11**
We paginate contracts on the smart-meter API side. It also gives us back an array of pages which the front-end can use for display purposes. 

The front-end isn't entirely hooked up yet, as we need some javascript glue to call successive pages once buttons are clicked.

### CODE REVIEW

File: client/meter/blockchain.js [hunk 1]
```diff
@@ -58,21 +59,35 @@ async function myBuyerContracts() {
...
       contracts.push({contractAddr: contractAddr, unitPrice: unitPrice, offeredAmount: offeredAmount})
+    } else {
+      // We always want to return req.limit items, so we increase the conditional of the loop
+      // if this contract is deregistered
+      req.offset++
     }
   }
-  return contracts;
+
+  pageCount = Math.ceil(count / req.query.limit)
```
Comment (Norbo11) [inline client/meter/blockchain.js]: This should actually be registeredCount, but we don't have that functionality yet, hence I added it to Master contract (not yet tested).

File: client/meter/blockchain.js [hunk 2]
```diff
@@ -58,21 +59,35 @@ async function myBuyerContracts() {
...
-      const unitPrice = await contract.unitPrice();
+      const contract = makeEnergyContract(contractAddr)
+      const offeredAmount = await contract.offeredAmount()
+      const unitPrice = await contract.unitPrice()
+
       contracts.push({contractAddr: contractAddr, unitPrice: unitPrice, offeredAmount: offeredAmount})
+    } else {
+      // We always want to return req.limit items, so we increase the conditional of the loop
+      // if this contract is deregistered
+      req.offset++
```
Comment (fangyi-zhou) [inline client/meter/blockchain.js]: Now i see why it's a problem

Comment (Norbo11) [reply client/meter/blockchain.js]: I think this is the first comment in our entire codebase

**Comment by fangyi-zhou** [comment]
Please cherry-pick the commits related to this PR

**Pull request closed without merging**